src/Streamly/Internal/Data/Parser/ParserK/Type.hs

```diff
@@ -415,21 +254,21 @@ instance Monad m => Fail.MonadFail (Parser m a) where
     fail = die
 #endif
 
-instance (MonadThrow m, MonadReader r m, MonadCatch m) => MonadReader r (Parser m a) where
+instance (MonadThrow m, MonadReader r m, MonadCatch m) =>
+    MonadReader r (Parser m a) where
     {-# INLINE ask #-}
     ask = fromEffect ask
-    {-# INLINE local #-}
-    local f (fromParserK -> dp) =
-      toParserK $ local f dp
 
+    {-# INLINE local #-}
+    local f (id -> dp) = local f dp
```

Comment:
This is an infinite loop, `local` is calling itself. We should actually implement it. See the MonadReader implementation of ParserD for example.